File: src/com/github/gtache/editor/EditorMouseMotionListenerImpl.scala
package com.github.gtache.editor

import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.editor.event.{EditorMouseEvent, EditorMouseMotionListener}

/**
  * Class listening for mouse movement in an editor (used for hover)
  */
class EditorMouseMotionListenerImpl extends EditorMouseMotionListener {

  private var manager: EditorEventManager = _
  private val LOG: Logger = Logger.getInstance(classOf[EditorMouseMotionListenerImpl])

  def setManager(manager: EditorEventManager): Unit = {
    this.manager = manager
  }

  override def mouseDragged(e: EditorMouseEvent): Unit = {}

  override def mouseMoved(e: EditorMouseEvent): Unit = {
    if (manager == null) {
      LOG.error("No manager")
    } else {
      manager.mouseMoved(e)
    }
  }
}

File: intellij-lsp/src/main/kotlin/com/github/gtache/lsp/editor/listeners/EditorMouseMotionListenerImpl.kt
package com.github.gtache.lsp.editor.listeners

import com.intellij.openapi.editor.event.EditorMouseEvent
import com.intellij.openapi.editor.event.EditorMouseMotionListener

/**
 * Class listening for mouse movement in an editor (used for hover)
 */
class EditorMouseMotionListenerImpl : EditorMouseMotionListener, AbstractLSPListener() {

    override fun mouseDragged(e: EditorMouseEvent): Unit {}

    override fun mouseMoved(e: EditorMouseEvent): Unit {
        if (checkEnabled()) {
            manager?.mouseMoved(e)
        }
    }
}